### PR TITLE
LPS-186127_6

### PR DESCRIPTION
**Pull request by brianwulbern**
https://issues.liferay.com/browse/LPS-186127

In an effort to provide context that is missing in ambiguous single-word language keys, I worked with the SF team to make a (for now, `default=false`) check on single-word keys, marking them as failed when they lack [context] brackets.

The commonly quoted example of an ambiguous single-word pain-point, is in 'order' - which can mean anything from a command, to how a list is sorted, to a purchase order.  And in each case, translations have difficulty working with this, because they would need to change the overarching translation for 'order' in Crowdin, but where other languages would use different words to say 'order', English would use the same in all cases.  With this change, we can have [order[buy], order[ranking(n)], and order[sort]](https://github.com/brianwulbern/liferay-portal/pull/429/files#diff-ef119629946c9b5876f3ddaf7e739fa3316781f38535fa10141db62509a5d684R11993-R11995).  You'll see I've contextualized some single-word keys, and we have a whitelist for keys that aren't ambiguous on their own, so they can skip the check.

After we can clean these keys up, I imagine we can roll this out to Engineering to stick to this formatting convention.  With all the single-word keys already added to source over the years, there won't be many more to add.

### CODE REVIEW

**Comment by brianwulbern** [comment]
ci:test:sf

**Comment by brianwulbern** [comment]
ci:test:relevant

**Comment by brianwulbern** [comment]
ci:test:stable

**Comment by brianwulbern** [comment]
I'm going to test it here one last time before forwarding it to Brian.

cc @ling-alan-huang

**Comment by ling-alan-huang** [comment]
@brianwulbern Missing language keys failures are related to this PR.
You have to fix the language keys in source code, or SF complains.

**Comment by brianwulbern** [comment]
ci:test:sf

**Comment by brianwulbern** [comment]
ci:test:sf

**Comment by brianwulbern** [comment]
ci:test:stable

**Comment by brianwulbern** [comment]
ci:test:relevant

**Comment by brianwulbern** [comment]
ci:test:sf

**Comment by brianwulbern** [comment]
ci:test:sf

**Comment by brianwulbern** [comment]
ci:test:stable

**Comment by brianwulbern** [comment]
@ling-alan-huang do you know how to resolve that test? https://github.com/brianwulbern/liferay-portal/pull/429/files#diff-bf13fae2c0d1c89bae6fd95a692aaae214b26521e27665e768c607e1c5d1e0f4R141-R143

I'm running into this:
```
Error: expect(element).toHaveTextContent()
Expected element to have text content:

/order[buy]0/i

Received:

order[buy]0 selectedselect-attributes
```

When running `../../../../gradlew packageRunTest` at `master-portal/modules/apps/analytics/analytics-settings-web`

I'm ready to just include an ambiguous key for 'order' just to avoid the error that I can't seem to resolve.

File: modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/attributes/__tests__/Attributes.spec.tsx [hunk 1]
```diff
@@ -107,10 +107,10 @@ describe('Attributes', () => {
 
 		expect(screen.getByText('products')).toBeInTheDocument();
 
-		expect(screen.getByText('order')).toBeInTheDocument();
+		expect(screen.getByText('order[buy]')).toBeInTheDocument();
```
Comment (ling-alan-huang) [inline modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/attributes/__tests__/Attributes.spec.tsx]: @brianwulbern I am not familiar with Front-end code, but I think the '**order**' here is not the key for language.
and this is the way how Front-end code uses language key.
`Liferay.Language.get('attribute[noun]')`

File: modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/attributes/__tests__/Attributes.spec.tsx [hunk 2]
```diff
@@ -107,10 +107,10 @@ describe('Attributes', () => {
...
 	});
 
-	it('renders Attributes with values for account, order, people, product', async () => {
+	it('renders Attributes with values for account, order[buy], people, product', async () => {
```
Comment (ling-alan-huang) [inline modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/attributes/__tests__/Attributes.spec.tsx]: this is not using the language key.

File: modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/attributes/__tests__/Attributes.spec.tsx (hunk 1 above)
Comment (brianwulbern) [reply modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/attributes/__tests__/Attributes.spec.tsx]: @ling-alan-huang ok, yeah, I'm trying out a way where I clear out that one, and the ones in `Attributes.tsx`

File: modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/attributes/__tests__/Attributes.spec.tsx [hunk 3]
```diff
@@ -138,7 +138,9 @@ describe('Attributes', () => {
 			/selected/i
 		);
 
-		expect(await screen.findByRole(/order/i)).toHaveTextContent(/order0/i);
+		expect(await screen.findByRole(/order/i)).toHaveTextContent(
```
Comment (ling-alan-huang) [inline modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/attributes/__tests__/Attributes.spec.tsx]: and here, we don't need to change here.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/test/js/pages/RuleList.es.js [hunk 4]
```diff
@@ -242,7 +242,7 @@ describe('RuleList', () => {
 				container.querySelector(
 					'.list-group .list-group-item .autofit-col'
 				).textContent
-			).toRule('if user user belongs-to list Guest require Foo');
+			).toRule('if user user belongs-to list[noun] Guest require Foo');
```
Comment (ling-alan-huang) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/test/js/pages/RuleList.es.js:245]: here?

File: modules/apps/layout/layout-content-page-editor-web/test/page_editor/app/plugins/fragments_and_widgets/components/FragmentsSidebar.test.js [hunk 5]
```diff
@@ -602,7 +602,7 @@ describe('FragmentsSidebar', () => {
 			userEvent.click(screen.getByTitle('switch-to-card-view'));
 
 			expect(
-				screen.getByTitle('switch-to-list-view')
+				screen.getByTitle('switch-to-list[noun]-view')
```
Comment (ling-alan-huang) [inline modules/apps/layout/layout-content-page-editor-web/test/page_editor/app/plugins/fragments_and_widgets/components/FragmentsSidebar.test.js:605]: here

File: modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/attributes/__tests__/Attributes.spec.tsx (hunk 3 above)
Comment (brianwulbern) [reply modules/apps/analytics/analytics-settings-web/src/main/resources/META-INF/resources/js/components/attributes/__tests__/Attributes.spec.tsx]: Yeah, I'm switching this one back

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/test/js/pages/RuleList.es.js (hunk 4 above)
Comment (brianwulbern) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/test/js/pages/RuleList.es.js:245]: This one got flagged by an SF check

File: modules/apps/layout/layout-content-page-editor-web/test/page_editor/app/plugins/fragments_and_widgets/components/FragmentsSidebar.test.js (hunk 5 above)
Comment (brianwulbern) [reply modules/apps/layout/layout-content-page-editor-web/test/page_editor/app/plugins/fragments_and_widgets/components/FragmentsSidebar.test.js:605]: Same as this one.

**Comment by brianwulbern** [comment]
ci:test:sf

**Comment by brianwulbern** [comment]
ci:test:stable

**Comment by brianwulbern** [comment]
ci:stop

**Comment by brianwulbern** [comment]
ci:stop

**Comment by ling-alan-huang** [comment]
SF fix is here: https://github.com/brianchandotcom/liferay-portal/pull/136632

**Comment by brianwulbern** [comment]
ci:test:sf

**Comment by brianwulbern** [comment]
ci:test:stable

**Comment by brianwulbern** [comment]
ci:reevaluate:1277401_8806

**Comment by brianwulbern** [comment]
ci:test:stable

**Comment by brianwulbern** [comment]
ci:forward

**Comment by brianwulbern** [comment]
ci:forward:force

**Comment by brianwulbern** [comment]
ci:stop

**Comment by brianwulbern** [comment]
ci:stop

**Comment by brianwulbern** [comment]
Analyzed the relevant suite test failures.  Nothing looks related to the changes.

**Pull request closed without merging**